src/CMakeLists.txt

```
@@ -48,25 +52,38 @@ function(gz_msgs_protoc)
   set(protoc_args)
   set(output_files)
 
-  foreach(proto_path ${gz_msgs_protoc_PROTO_PATH})
-    list(APPEND protoc_args "--proto_path=${proto_path}")
-  endforeach()
-
   set(proto_package_dir ".")
   if(gz_msgs_protoc_PROTO_PACKAGE)
     string(REPLACE "." "/" proto_package_dir ${gz_msgs_protoc_PROTO_PACKAGE})
   endif()
 
   if(gz_msgs_protoc_GENERATE_CPP)
+    # Full path to generated header (${PROJECT_BINARY_DIR}/include/gz/msgs/foo.pb.h)
     set(output_header "${gz_msgs_protoc_OUTPUT_CPP_DIR}${proto_package_dir}/${FIL_WE}.pb.h")
+    # Full path to generated detail header (${PROJECT_BINARY_DIR}/include/gz/msgs/detail/foo.pb.h)
+    set(output_detail_header "${gz_msgs_protoc_OUTPUT_CPP_DIR}${proto_package_dir}/detail/${FIL_WE}.pb.h")
+    # Full path to generated ignition header (${PROJECT_BINARY_DIR}/include/ignition/msgs/foo.pb.h)
+    set(output_ign_header "${gz_msgs_protoc_OUTPUT_CPP_DIR}/ignition/msgs/${FIL_WE}.pb.h")
+    # Full path to generated ignition header (${PROJECT_BINARY_DIR}/include/foo.pb.cc)
```

do we need to change paths to `gz` too ?

```suggestion
    # Full path to generated GAZEBO header (${PROJECT_BINARY_DIR}/include/ignition/msgs/foo.pb.h)
    set(output_gz_header "${gz_msgs_protoc_OUTPUT_CPP_DIR}/ignition/msgs/${FIL_WE}.pb.h")
    # Full path to generated ignition header (${PROJECT_BINARY_DIR}/include/foo.pb.cc)
```